src/mc-schema-broker.c

```diff
@@ -860,60 +858,56 @@ static bool insert_encryptionInformation(const mc_schema_broker_t *sb,
+        // crypt_shared and the server expect "encryptionInformation" nested in the "explain" document. Example:
+        // {
+        //    "explain": {
+        //       "find": "to-crypt_shared-or-server"
+        //       "encryptionInformation": {}
+        //    }
+        // }
+        // mongocryptd expects "encryptionInformation" to be a sibling of the "explain" document. Example:
+        // {
+        //    "explain": { "find": "to-mongocryptd" },
+        //    "encryptionInformation": {}
+        // }
+        BSON_ASSERT(bson_iter_init_find(&iter, cmd, "explain"));
+        if (!BSON_ITER_HOLDS_DOCUMENT(&iter)) {
+            CLIENT_ERR("expected 'explain' to be document");
             goto fail;
         }
-        bson_destroy(&out);
-        goto success;
-    }
 
-    // The "explain" command for csfle is a special case.
-    // mongocryptd expects "encryptionInformation" to be a sibling of the
-    // "explain" document. Example:
-    // {
-    //    "explain": { "find": "to-mongocryptd" },
-    //    "encryptionInformation": {}
-    // }
-    // csfle and mongod expect "encryptionInformation" to be nested in the
-    // "explain" document. Example:
-    // {
-    //    "explain": {
-    //       "find": "to-csfle-or-mongod"
-    //       "encryptionInformation": {}
-    //    }
-    // }
-    BSON_ASSERT(bson_iter_init_find(&iter, cmd, "explain"));
-    if (!BSON_ITER_HOLDS_DOCUMENT(&iter)) {
-        CLIENT_ERR("expected 'explain' to be document");
-        goto fail;
-    }
+        {
+            bson_t tmp;
+            if (!mc_iter_document_as_bson(&iter, &tmp, status)) {
+                goto fail;
+            }
+            bson_destroy(&explain);
+            bson_copy_to(&tmp, &explain);
+        }
 
-    {
-        bson_t tmp;
-        if (!mc_iter_document_as_bson(&iter, &tmp, status)) {
+        if (!append_encryptionInformation(sb, cmd_name, &explain, status)) {
             goto fail;
         }
-        bson_destroy(&explain);
-        bson_copy_to(&tmp, &explain);
-    }
 
-    if (!append_encryptionInformation(sb, cmd_name, &explain, status)) {
-        goto fail;
-    }
+        if (!BSON_APPEND_DOCUMENT(&out, "explain", &explain)) {
+            CLIENT_ERR("unable to append 'explain' document");
+            goto fail;
+        }
 
-    if (!BSON_APPEND_DOCUMENT(&out, "explain", &explain)) {
-        CLIENT_ERR("unable to append 'explain' document");
-        goto fail;
+        bson_copy_to_excluding_noinit(cmd, &out, "explain", NULL);
+        bson_destroy(cmd);
+        if (!bson_steal(cmd, &out)) {
+            CLIENT_ERR("failed to steal BSON with encryptionInformation");
+            goto fail;
+        }
+        goto success;
     }
 
-    bson_copy_to_excluding_noinit(cmd, &out, "explain", NULL);
-    bson_destroy(cmd);
-    if (!bson_steal(cmd, &out)) {
-        CLIENT_ERR("failed to steal BSON with encryptionInformation");
+    // Default case: append "encryptionInformation" at top-level.
```

Comment:
```suggestion
    // Default case: append "encryptionInformation" at top-level. Example:
    // {
    //    "<command name>": { ... }
    //    "encryptionInformation": {}
    // }
```
Recommend documenting an example of the default case (top-level "encryptionInformation") here instead of within the explain-command-specific blcok above.